modules/index/src/main/java/org/apache/ignite/internal/index/IndexBuilder.java

```diff
@@ -133,71 +176,77 @@ private BuildIndexTask(
 
         @Override
         public void run() {
-            if (!busyLock.enterBusy()) {
-                return;
-            }
+            completedFuture(null)
+                    .thenCompose(unused -> inBusyLock(() -> {
+                        // At the time of creating the index, we should have already waited for the table to be created and its raft of
+                        // clients (services) to start for all partitions, so there should be no errors.
+                        RaftGroupService raftGroupService = table.internalTable().partitionRaftGroupService(partitionId);
+
+                        return raftGroupService
+                                // We do not check the presence of nodes in the topology on purpose, so as not to get into races on
+                                // rebalancing, it will be more convenient and reliable for us to wait for a stable topology with a
+                                // chosen leader.
```

Comment:
I don't get it, maybe you can paraphrase it somehow?

Reply:
Try fix it.